lib/internal/url.js

```diff
@@ -487,7 +487,7 @@ Object.defineProperties(URL.prototype, {
       if (!search) {
         ctx.query = null;
         ctx.flags &= ~binding.URL_FLAGS_HAS_QUERY;
-        this[searchParams][searchParams] = {};
+        this[searchParams][searchParams] = [];
```

Comment:
@joyeecheung, done, PTAL.

This does bring a drop to performance when `!search`. To counter this problem I added a few conditions to avoid parsing as much as possible.

After those changes, when [alternative](https://github.com/nodejs/node/blob/6ca1bdfeb158aa8a4b981b715278b30c4a019ff0/benchmark/url/whatwg-url-properties.js#L54) is changed to `''`, there is a 2.8% drop in performance; but when it is `'?'` there is a 400% *increase*, as measured by 

```sh
node benchmark/compare.js --new ./after --old ./orig \
  --set n=1e5 --set prop=search --set 'url=http://example.com/' \
  --filter properties url
```

That's pretty good IMO.